src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapSessionOptions.Linux.cs

```diff
@@ -16,5 +18,52 @@ public int ProtocolVersion
             get => GetPtrValueHelper(LdapOption.LDAP_OPT_VERSION).ToInt32();
             set => SetPtrValueHelper(LdapOption.LDAP_OPT_VERSION, new IntPtr(value));
         }
+
+        public ReferralChasingOptions ReferralChasing
+        {
+            get
+            {
+                return GetBoolValueHelper(LdapOption.LDAP_OPT_REFERRALS) ? ReferralChasingOptions.All : ReferralChasingOptions.None;
+            }
+            set
+            {
+                if (((value) & (~ReferralChasingOptions.All)) != 0)
+                {
+                    throw new InvalidEnumArgumentException(nameof(value), (int)value, typeof(ReferralChasingOptions));
+                }
+                if (!(value == ReferralChasingOptions.None || value == ReferralChasingOptions.All))
```

Comment:
Nit: I personally find `a != x && b != y` easier to read than `!(a == x || b == y)`